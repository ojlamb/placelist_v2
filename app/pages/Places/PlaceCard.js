import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';


const styles = {
  card: {
    maxWidth: 345
  },
  media: {
    height: 140
  },
  link: {
    textDecoration: 'none',
  }
};


const getCardImage = data => require(`common/images/${data}.jpg`);


export class PlaceCard extends React.PureComponent {
  render() {
    const { classes } = this.props;
    return (
      <Card className={classes.card}>
        <CardMedia
          className={classes.media}
          image={getCardImage(this.props.category)}
          title={this.props.name}
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {this.props.name}
          </Typography>
          <Typography component="p">{this.props.address}</Typography>
        </CardContent>
        <CardActions>
          <Link className={classes.link} to={`/place/${this.props.id}`}>
            <Button size="small" color="primary">
              More
            </Button>
          </Link>
        </CardActions>
      </Card>
    );
  }
}
PlaceCard.propTypes = {
  address: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired
};

export default withStyles(styles)(PlaceCard);
