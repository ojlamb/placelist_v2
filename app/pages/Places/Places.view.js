import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Grid from '@material-ui/core/Grid';
import { withStyles } from '@material-ui/core/styles';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import PlaceCard from './PlaceCard';
import NewPlace from '../Components/NewPlace';

export const styles = theme => ({
  actionButton: {
    margin: 0,
    top: 'auto',
    right: 20,
    bottom: 20,
    left: 'auto',
    position: 'fixed'
  },
  icon: {
    marginRight: theme.spacing.unit * 2
  },
  layout: {
    width: 'auto',
    margin: '10px',
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(1100 + theme.spacing.unit * 3 * 2)]: {
      width: 1100,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  cardGrid: {
    padding: `${theme.spacing.unit * 8}px 0`
  }
});

export class PlaceView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      places: [],
      formOpen: false
    };
    this.createPlace = this.createPlace.bind(this);
    this.handleClose = this.handleClose.bind(this);
  }

  componentDidMount() {
    this.props.actions.loadPlaces();
  }

  handleClickOpen = () => {
    this.setState({ formOpen: true });
  };

  handleClose = () => {
    this.setState({ formOpen: false });
  };

  createPlace = (place) => {
    const { actions } = this.props;
    actions.createPlace(place);
    this.handleClose();
  };

  render() {
    const { classes } = this.props;
    return (
      <div>
        <Fab
          color="primary"
          aria-label="Add"
          className={classNames(classes.button, classes.actionButton)}
          onClick={this.handleClickOpen}
        >
          <AddIcon />
        </Fab>
        <NewPlace {...this.state} createPlace={this.createPlace} handleClose={this.handleClose} />
        <div
          style={{ maxHeight: '100vh', overflow: 'auto' }}
        >
          <div className={classNames(classes.layout, classes.cardGrid)}>
            <Grid container spacing={32}>
              {this.props.places.map((place, i) => (
                <Grid item key={i} sm={6} md={4} lg={3}>
                  <PlaceCard {...place} key={place.id} />
                </Grid>
              ))}
            </Grid>
          </div>
        </div>
      </div>
    );
  }
}

PlaceView.propTypes = {
  actions: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
  places: PropTypes.array.isRequired,
};

export default withStyles(styles)(PlaceView);
