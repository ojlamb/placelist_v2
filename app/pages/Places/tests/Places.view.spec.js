import React from 'react';
import theme from 'test/theme';
import { PlaceView, styles } from '../Places.view';

const getProps = ({
  places
} = {}) => ({
  classes: {},
  places,
  actions: {
    loadPlaces: jest.fn(),
    createPlace: jest.fn()
  }
});

const getJSX = props => (<PlaceView {...getProps(props)} />);

describe('PlaceView', () => {
  describe('styles', () => {
    it('matches snapshot', () => {
      expect(styles(theme)).toMatchSnapshot();
    });
  });

  describe('snapshots', () => {
    const scenarios = [
      { places: [{ id: 1 }, { id: 2 }] }
    ];

    scenarios.forEach((props, i) => {
      describe(`Scenario ${i + 1}`, () => {
        it('matches snapshot', () => {
          const wrapper = shallow(getJSX(props));
          expect(wrapper).toMatchSnapshot();
        });
      });
    });
  });

  describe('methods', () => {
    let wrapper;
    let instance;
    beforeAll(() => {
      const props = { places: [{ id: 1 }, { id: 2 }] };
      wrapper = shallow(<PlaceView {...getProps(props)} />);
      instance = wrapper.instance();
    });
    it('updates the state for open form', () => {
      instance.handleClickOpen();
      expect(wrapper.state('formOpen')).toBe(true);
    });
    it('updates the state for close form', () => {
      instance.handleClose();
      expect(wrapper.state('formOpen')).toBe(false);
    });
    it('updates the state for close form', () => {
      instance.createPlace({ id: 1 });
      expect(wrapper.state('formOpen')).toBe(false);
    });
  });
});
