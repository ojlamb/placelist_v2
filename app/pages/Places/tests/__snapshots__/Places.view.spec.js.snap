// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceView snapshots Scenario 1 matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlaceView
    actions={
      Object {
        "createPlace": [MockFunction],
        "loadPlaces": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
    classes={Object {}}
    places={
      Array [
        Object {
          "id": 1,
        },
        Object {
          "id": 2,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Fab)
          aria-label="Add"
          className=""
          color="primary"
          onClick={[Function]}
        >
          <pure(AddIcon) />
        </WithStyles(Fab)>,
        <NewPlace
          createPlace={[Function]}
          formOpen={false}
          handleClose={[Function]}
          places={Array []}
        />,
        <div
          style={
            Object {
              "maxHeight": "100vh",
              "overflow": "auto",
            }
          }
        >
          <div
            className=""
          >
            <WithStyles(Grid)
              container={true}
              spacing={32}
            >
              <WithStyles(Grid)
                item={true}
                lg={3}
                md={4}
                sm={6}
              >
                <WithStyles(PlaceCard)
                  id={1}
                />
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                lg={3}
                md={4}
                sm={6}
              >
                <WithStyles(PlaceCard)
                  id={2}
                />
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-label": "Add",
          "children": <pure(AddIcon) />,
          "className": "",
          "color": "primary",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "createPlace": [Function],
          "formOpen": false,
          "handleClose": [Function],
          "places": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className=""
          >
            <WithStyles(Grid)
              container={true}
              spacing={32}
            >
              <WithStyles(Grid)
                item={true}
                lg={3}
                md={4}
                sm={6}
              >
                <WithStyles(PlaceCard)
                  id={1}
                />
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                lg={3}
                md={4}
                sm={6}
              >
                <WithStyles(PlaceCard)
                  id={2}
                />
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </div>,
          "style": Object {
            "maxHeight": "100vh",
            "overflow": "auto",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(Grid)
              container={true}
              spacing={32}
            >
              <WithStyles(Grid)
                item={true}
                lg={3}
                md={4}
                sm={6}
              >
                <WithStyles(PlaceCard)
                  id={1}
                />
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                lg={3}
                md={4}
                sm={6}
              >
                <WithStyles(PlaceCard)
                  id={2}
                />
              </WithStyles(Grid)>
            </WithStyles(Grid)>,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Grid)
                  item={true}
                  lg={3}
                  md={4}
                  sm={6}
                >
                  <WithStyles(PlaceCard)
                    id={1}
                  />
                </WithStyles(Grid)>,
                <WithStyles(Grid)
                  item={true}
                  lg={3}
                  md={4}
                  sm={6}
                >
                  <WithStyles(PlaceCard)
                    id={2}
                  />
                </WithStyles(Grid)>,
              ],
              "container": true,
              "spacing": 32,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(PlaceCard)
                    id={1}
                  />,
                  "item": true,
                  "lg": 3,
                  "md": 4,
                  "sm": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "id": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(PlaceCard)
                    id={2}
                  />,
                  "item": true,
                  "lg": 3,
                  "md": 4,
                  "sm": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "class",
                  "props": Object {
                    "id": 2,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Fab)
            aria-label="Add"
            className=""
            color="primary"
            onClick={[Function]}
          >
            <pure(AddIcon) />
          </WithStyles(Fab)>,
          <NewPlace
            createPlace={[Function]}
            formOpen={false}
            handleClose={[Function]}
            places={Array []}
          />,
          <div
            style={
              Object {
                "maxHeight": "100vh",
                "overflow": "auto",
              }
            }
          >
            <div
              className=""
            >
              <WithStyles(Grid)
                container={true}
                spacing={32}
              >
                <WithStyles(Grid)
                  item={true}
                  lg={3}
                  md={4}
                  sm={6}
                >
                  <WithStyles(PlaceCard)
                    id={1}
                  />
                </WithStyles(Grid)>
                <WithStyles(Grid)
                  item={true}
                  lg={3}
                  md={4}
                  sm={6}
                >
                  <WithStyles(PlaceCard)
                    id={2}
                  />
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-label": "Add",
            "children": <pure(AddIcon) />,
            "className": "",
            "color": "primary",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "createPlace": [Function],
            "formOpen": false,
            "handleClose": [Function],
            "places": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className=""
            >
              <WithStyles(Grid)
                container={true}
                spacing={32}
              >
                <WithStyles(Grid)
                  item={true}
                  lg={3}
                  md={4}
                  sm={6}
                >
                  <WithStyles(PlaceCard)
                    id={1}
                  />
                </WithStyles(Grid)>
                <WithStyles(Grid)
                  item={true}
                  lg={3}
                  md={4}
                  sm={6}
                >
                  <WithStyles(PlaceCard)
                    id={2}
                  />
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </div>,
            "style": Object {
              "maxHeight": "100vh",
              "overflow": "auto",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <WithStyles(Grid)
                container={true}
                spacing={32}
              >
                <WithStyles(Grid)
                  item={true}
                  lg={3}
                  md={4}
                  sm={6}
                >
                  <WithStyles(PlaceCard)
                    id={1}
                  />
                </WithStyles(Grid)>
                <WithStyles(Grid)
                  item={true}
                  lg={3}
                  md={4}
                  sm={6}
                >
                  <WithStyles(PlaceCard)
                    id={2}
                  />
                </WithStyles(Grid)>
              </WithStyles(Grid)>,
              "className": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Grid)
                    item={true}
                    lg={3}
                    md={4}
                    sm={6}
                  >
                    <WithStyles(PlaceCard)
                      id={1}
                    />
                  </WithStyles(Grid)>,
                  <WithStyles(Grid)
                    item={true}
                    lg={3}
                    md={4}
                    sm={6}
                  >
                    <WithStyles(PlaceCard)
                      id={2}
                    />
                  </WithStyles(Grid)>,
                ],
                "container": true,
                "spacing": 32,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(PlaceCard)
                      id={1}
                    />,
                    "item": true,
                    "lg": 3,
                    "md": 4,
                    "sm": 6,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "1",
                    "nodeType": "class",
                    "props": Object {
                      "id": 1,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(PlaceCard)
                      id={2}
                    />,
                    "item": true,
                    "lg": 3,
                    "md": 4,
                    "sm": 6,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "2",
                    "nodeType": "class",
                    "props": Object {
                      "id": 2,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PlaceView styles matches snapshot 1`] = `
Object {
  "actionButton": Object {
    "bottom": 20,
    "left": "auto",
    "margin": 0,
    "position": "fixed",
    "right": 20,
    "top": "auto",
  },
  "cardGrid": Object {
    "padding": "80px 0",
  },
  "icon": Object {
    "marginRight": 20,
  },
  "layout": Object {
    "margin": "10px",
    "marginLeft": 30,
    "marginRight": 30,
    "undefined": Object {
      "marginLeft": "auto",
      "marginRight": "auto",
      "width": 1100,
    },
    "width": "auto",
  },
}
`;
